build-system/tasks/runtime-test/helpers.js

```diff
@@ -242,48 +245,23 @@ function karmaRunStart_() {
 }
 
 /**
- * Runs tests in sauce labs
+ * Runs tests in Sauce Labs in batches.
  *
- * If --stable is provided, runs tests only on stable browsers in sauce labs without batching.
- * If --beta is provided, runs tests only on beta browsers in sauce labs without batching. Does not fail.
- * If neither --stable nor --beta are provided, runs test on all browsers in sauce labs with batching.
+ * If --stable is provided, runs tests only on stable browsers in Sauce Labs.
+ * If --beta is provided, runs tests only on beta browsers in Sauce Labs. Does not fail the build.
+ * If neither --stable nor --beta are provided, runs test on all browsers in Sauce Labs.
  *
  * @param {Object} config karma config
  * @return {!Promise<number>} exitCode
  */
 async function runTestInSauceLabs(config) {
-  const browsers = {stable: [], beta: []};
-  for (const browserId of config.browsers) {
-    browsers[
-      browserId.toLowerCase().endsWith('_beta') ? 'beta' : 'stable'
-    ].push(browserId);
-  }
-
-  if (argv.stable) {
-    config.browsers = browsers.stable;
-    return createKarmaServer(config, reportTestRunComplete);
-  }
-
-  if (argv.beta) {
-    config.browsers = browsers.beta;
-    const betaExitCode = await createKarmaServer(config, reportTestRunComplete);
-    if (betaExitCode != 0) {
-      log(
-        yellow('Some tests have failed on'),
-        cyan('beta'),
-        yellow('browsers.')
-      );
-      log(
-        yellow('This is not currently a fatal error, but will become an'),
-        yellow('error once the beta browsers are released as next stable'),
-        yellow('version!')
-      );
-    }
-
-    return 0;
-  }
+  const isBeta = browserId => browserId.toLowerCase().endsWith('_beta');
+  const isStable = browserId => !isBeta(browserId);
 
-  return await runTestInBatches_(config, browsers);
+  return await runTestInBatches_(config, {
+    beta: argv.stable ? [] : config.browsers.filter(isBeta),
+    stable: argv.beta ? [] : config.browsers.filter(isStable),
```

Comment:
Can we flip this so that stable browsers are set when argv.stable is set? And if neither flags are present, default to all or stable, up to you.

Reply:
Done, PTAL